packages/guidecue/src/Guidecue.spec.tsx

```diff
@@ -0,0 +1,394 @@
+      title={guidecueTitle}
+      refEl={ref}
+      buttonText={buttonText}
+      currentStep={currentStep}
+      numberOfSteps={numberOfSteps}
+      {...props}
+    >
+      <div>{guidecueChildren}</div>
+    </Guidecue>
+  );
+};
+
+function renderGuidecue(
+  props: Partial<React.ComponentProps<typeof Guidecue>> = {},
+) {
+  const utils = render(
+    <>
+      <div data-testid="backdrop" />
+      <GuidecueWrapper {...props} />
+    </>,
+  );
+
+  const backdrop = utils.getByTestId('backdrop');
+
+  return { ...utils, backdrop };
+}
+
+describe('packages/guidecue', () => {
+  describe('A11y', () => {
+    test('does not have basic accessibility violations', async () => {
+      const { container } = renderGuidecue({ open: true });
+      const results = await axe(container);
+      expect(results).toHaveNoViolations();
+    });
+  });
+
+  describe('Stand-alone tooltip', () => {
+    test('is visible when open is "true"', async () => {
+      const { getByRole } = renderGuidecue({ open: true });
+      const guidecue = getByRole('dialog');
+      await waitFor(() => expect(guidecue).toBeVisible());
+    });
+
+    test('closes when the primary button is clicked', async () => {
+      const onPrimaryButtonClick = jest.fn();
+      const { getByRole } = renderGuidecue({
+        open: true,
+        onPrimaryButtonClick,
+      });
+      const guidecue = getByRole('dialog');
+      const button = getByRole('button');
+      userEvent.click(button);
+      await waitForElementToBeRemoved(guidecue);
+      expect(guidecue).not.toBeInTheDocument();
+    });
+
+    test('fires the onPrimaryButtonClick handler once when the primary button is clicked', async () => {
+      const onPrimaryButtonClick = jest.fn();
+      const { getByRole } = renderGuidecue({
+        open: true,
+        onPrimaryButtonClick,
+      });
+      const button = getByRole('button');
+      userEvent.click(button);
+      await act(async () => {
+        await waitForTimeout(100);
+        expect(onPrimaryButtonClick).toHaveBeenCalledTimes(1);
+      });
+    });
+
+    test('a click ourside of the tooltip should do nothing', async () => {
+      const { getByRole, backdrop } = renderGuidecue({
+        open: true,
+      });
+
+      const guidecue = getByRole('dialog');
+      userEvent.click(backdrop);
+
+      await expectElementToNotBeRemoved(guidecue);
+      expect(guidecue).toBeVisible();
+    });
+
+    test('closes guidecue when escape key is pressed', async () => {
+      const { getByRole } = renderGuidecue({ open: true });
+
+      const guidecue = getByRole('dialog');
+      userEvent.type(guidecue, '{esc}');
+
+      await waitForElementToBeRemoved(guidecue);
+      expect(guidecue).not.toBeInTheDocument();
+    });
+
+    test('content should not be in the container', () => {
+      const { container, getByTestId } = renderGuidecue({
+        open: true,
+      });
+      expect(container).not.toContain(getByTestId(guidecueTestId));
+    });
```

Comment:
yeah this is unclear, I believe I was testing that the tooltip was not rendered inside the guidecue component because it should render in a portal. I think it should be a separate test and I should change the name to `content should render in a portal`